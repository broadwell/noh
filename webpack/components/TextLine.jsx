import React from "react";
import PropTypes from "prop-types";

const Line = props => (
  <div
    className={
      props.active ? "transcription__line active" : "transcription__line"
    }
  >
    <p className="transcription__original">{props.transcription}</p>
    <p className="transcription__translation">{props.translation}</p>
  </div>
);

Line.propTypes = {
  transcription: PropTypes.string.isRequired,
  translation: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired
};

export default Line;
