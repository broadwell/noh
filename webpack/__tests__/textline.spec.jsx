import React from "react";
import { shallow } from "enzyme";
import TextLine from "../components/TextLine";

describe("<TextLine>", () => {
  it("renders as expected with active class", () => {
    const component = shallow(
      <TextLine
        active
        translation="sample string"
        transcription="another sample string"
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders as expected without active class", () => {
    const component = shallow(
      <TextLine
        active={false}
        translation="sample string"
        transcription="another sample string"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
