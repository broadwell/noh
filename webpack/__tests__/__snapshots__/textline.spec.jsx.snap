// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLine> renders as expected with active class 1`] = `
<div
  className="transcription__line active"
>
  <p
    className="transcription__original"
  >
    another sample string
  </p>
  <p
    className="transcription__translation"
  >
    sample string
  </p>
</div>
`;

exports[`<TextLine> renders as expected without active class 1`] = `
<div
  className="transcription__line"
>
  <p
    className="transcription__original"
  >
    another sample string
  </p>
  <p
    className="transcription__translation"
  >
    sample string
  </p>
</div>
`;
